### PR TITLE
ft: proper versioning action for Vault

### PR DESCRIPTION
**Pull request by nicolas2bert**
S3C-156
For requests that have version id's use correct "action" when sending requestContext to vault.

### CODE REVIEW

**Comment by nicolas2bert** [comment]
E2E: http://ci.ironmann.io/gh/scality/Integration/11360

File: lib/api/apiUtils/authorization/prepareRequestContexts.js [hunk 1]
```diff
@@ -25,8 +28,15 @@ export default function prepareRequestContexts(apiMethod, request, sourceBucket,
...
+        apiMethodAfterVersionCheck = `${apiMethod}Version`;
+    } else {
+        apiMethodAfterVersionCheck = apiMethod;
+    }
     const requestContexts = [];
-    if (apiMethod === 'objectCopy' || apiMethod === 'objectPutCopyPart') {
+    if (apiMethodAfterVersionCheck === 'objectCopy' ||
+    apiMethodAfterVersionCheck === 'objectPutCopyPart') {
         // TODO: Categorize objectCopy requests with copy source version
         // ids as 'objectGetVersion' action (after related PR is merged)
```
Comment (electrachong) [inline lib/api/apiUtils/authorization/prepareRequestContexts.js]: add a check if `sourceVersionId !== undefined`, `'objectGet'` below (line 47) should become `objectGetVersion` api method.

Comment (electrachong) [inline lib/api/apiUtils/authorization/prepareRequestContexts.js]: and you can delete this comment afterward :)

**Comment by nicolas2bert** [comment]
E2E: http://ci.ironmann.io/gh/scality/Integration/11374

**Pull request merged**